x-pack/plugins/infra/public/components/asset_details/tabs/overview/metrics/chart_utils.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MultiValueClickContext } from '@kbn/embeddable-plugin/public';
+import { DatatableColumn, DatatableRow } from '@kbn/expressions-plugin/common/expression_types';
+
+type ChartClickContextData = MultiValueClickContext['data']['data'];
+
+export function isChartClickContextData(data: unknown): data is ChartClickContextData {
+  return (
+    Array.isArray(data) &&
+    data.length > 0 &&
+    Array.isArray(data[0].cells) &&
+    data[0].cells.length > 0
+  );
```

Comment:
This could be simplified. Wdyt?

```suggestion
  return Array.isArray(data?.[0]?.cells) && data[0].cells.length > 0;
```

Reply:
Agree, less verbose 👍

Reply:
Had to leave the `Array.isArray(data)` because it turns out TS does not allow `data?` in case `data: unknown`, bus combined second and third condition into one.